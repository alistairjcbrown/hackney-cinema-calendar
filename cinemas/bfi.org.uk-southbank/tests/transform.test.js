const fs = require("node:fs");
const path = require("node:path");
const transform = require("../transform");

const readMockData = (filename) =>
  fs.readFileSync(
    path.resolve(__dirname, `./mock-data/${filename}.txt`),
    "utf8",
  );

const moviesWithLocalScreenings = readMockData("movie-with-local-screenings");

jest.useFakeTimers().setSystemTime(new Date("2024-10-20"));

// Disable caching for test
jest.mock("../../../common/cache", () => ({
  dailyCache: (key, retrieve) => retrieve(),
}));

// Mock retrieve calls for additional data
jest.mock("../retrieve", () => (url) => {
  const [, slug] = url.split("article/");
  if (slug) {
    return readMockData(`show-${slug}`);
  }

  const [, id] = url.split("article_id=");
  const [, page] = url.match(/current_page=(\d)+&/);
  switch (id) {
    case "CE6463D3-DB5E-4E11-8A61-3D7F70969FA3": {
      return readMockData(`show-watership-down-${page}`);
    }
  }
});

describe("BFI Southbank", () => {
  describe("when no movies provided", () => {
    it("returns an empty list of events", async () => {
      const value = "";
      expect(await transform(value)).toEqual([]);
    });
  });

  describe("when movies with local screenings provided", () => {
    it("returns a list of events", async () => {
      const value = moviesWithLocalScreenings;
      expect(await transform(value)).toMatchSnapshot();
    });
  });
});
