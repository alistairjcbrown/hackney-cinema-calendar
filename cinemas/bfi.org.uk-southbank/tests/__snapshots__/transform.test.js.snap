// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BFI Southbank when movies with local screenings provided returns a list of events 1`] = `
[
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 5400000,
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/vera-end-of-an-era-qanda",
        "notes": "",
        "screen": "NFT1",
        "time": 1730830200000,
      },
    ],
    "title": "Vera: End of an Era",
    "url": "https://whatson.bfi.org.uk/Online/article/vera-end-of-an-era-qanda",
  },
  {
    "overview": {
      "actors": [
        "Lee Eol",
        "Park Won-sang",
        "Hwang Jung-min",
        "Ryoo Seung-bum",
      ],
      "categories": [],
      "directors": [
        "Yim Soon-rye",
      ],
      "duration": 6540000,
      "year": "2001",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/victory-qanda",
        "notes": "",
        "screen": "NFT2",
        "time": 1731703500000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/victory-qanda",
        "notes": "",
        "screen": "NFT2",
        "time": 1732903200000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/victory-qanda",
        "notes": "",
        "screen": "NFT1",
        "time": 1730918700000,
      },
    ],
    "title": "Waikiki BrothersLondon Korean Film Festival Opening Gala: Victory",
    "url": "https://whatson.bfi.org.uk/Online/article/victory-qanda",
  },
  {
    "overview": {
      "actors": [
        "Lee Eol",
        "Park Won-sang",
        "Hwang Jung-min",
        "Ryoo Seung-bum",
      ],
      "categories": [],
      "directors": [
        "Yim Soon-rye",
      ],
      "duration": 6540000,
      "year": "2001",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/waikiki-brothers",
        "notes": "",
        "screen": "NFT2",
        "time": 1731703500000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/waikiki-brothers",
        "notes": "",
        "screen": "NFT2",
        "time": 1732903200000,
      },
    ],
    "title": "Waikiki Brothers",
    "url": "https://whatson.bfi.org.uk/Online/article/waikiki-brothers",
  },
  {
    "overview": {
      "actors": [],
      "age-restriction": "12A",
      "categories": [],
      "directors": [],
      "duration": 6060000,
      "year": "2010",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/war-you-dont-see",
        "notes": "",
        "screen": "NFT1",
        "time": 1730147700000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/war-you-dont-see",
        "notes": "",
        "screen": "NFT2",
        "time": 1731780600000,
      },
    ],
    "title": "The War You Don’t See",
    "url": "https://whatson.bfi.org.uk/Online/article/war-you-dont-see",
  },
  {
    "overview": {
      "actors": [
        "voices of John Hurt",
        "Richard Briers",
        "Ralph Richardson",
      ],
      "age-restriction": "PG (suitable for ages 8+)",
      "categories": [],
      "directors": [
        "Martin Rosen",
      ],
      "duration": 5520000,
      "year": "1978",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/watership-down",
        "notes": "",
        "screen": "NFT4",
        "time": 1730211000000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/watership-down",
        "notes": "",
        "screen": "NFT4",
        "time": 1730290800000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/watership-down",
        "notes": "",
        "screen": "NFT4",
        "time": 1730385600000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/watership-down",
        "notes": "",
        "screen": "NFT4",
        "time": 1730464200000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/watership-down",
        "notes": "",
        "screen": "NFT3",
        "time": 1730550600000,
      },
    ],
    "title": "Watership Down",
    "url": "https://whatson.bfi.org.uk/Online/article/watership-down",
  },
  {
    "overview": {
      "actors": [
        "Moon Jung-suk",
        "Kim Seung-ho",
        "Kim Seok-hun",
        "Yu Gye-seon",
      ],
      "categories": [],
      "directors": [
        "Hong Eun-won",
      ],
      "duration": 5160000,
      "year": "1962",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/woman-judge-intro",
        "notes": "",
        "screen": "NFT2",
        "time": 1730139000000,
      },
    ],
    "title": "A Woman Judge",
    "url": "https://whatson.bfi.org.uk/Online/article/woman-judge-intro",
  },
  {
    "overview": {
      "actors": [
        "Moon Jung-suk",
        "Kim Seung-ho",
        "Kim Seok-hun",
        "Yu Gye-seon",
      ],
      "categories": [],
      "directors": [
        "Hong Eun-won",
      ],
      "duration": 5160000,
      "year": "1962",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/woman-judge",
        "notes": "",
        "screen": "NFT2",
        "time": 1731156000000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/woman-judge",
        "notes": "",
        "screen": "NFT3",
        "time": 1731963000000,
      },
    ],
    "title": "A Woman Judge",
    "url": "https://whatson.bfi.org.uk/Online/article/woman-judge",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [],
      "duration": 5400000,
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/womens-voices-forum",
        "notes": "",
        "screen": "NFT2",
        "time": 1731348600000,
      },
    ],
    "title": "Women’s Voices Forum",
    "url": "https://whatson.bfi.org.uk/Online/article/womens-voices-forum",
  },
  {
    "overview": {
      "actors": [],
      "categories": [],
      "directors": [
        "David Munro",
      ],
      "duration": 3180000,
      "year": "1979",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/year-zero-breaking-silence",
        "notes": "",
        "screen": "NFT2",
        "time": 1731953400000,
      },
    ],
    "title": "Year Zero: The Silent Death of Cambodia + Breaking the Silence: Truth and Lies in the War on Terror",
    "url": "https://whatson.bfi.org.uk/Online/article/year-zero-breaking-silence",
  },
  {
    "overview": {
      "actors": [
        "Michelle Yeoh",
        "Cynthia Rothrock",
      ],
      "age-restriction": "18",
      "categories": [],
      "directors": [
        "Corey Yuen",
      ],
      "duration": 5580000,
      "year": "1985",
    },
    "performances": [
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/yes-madam",
        "notes": "",
        "screen": "NFT2",
        "time": 1729626000000,
      },
      {
        "bookingUrl": "https://whatson.bfi.org.uk/Online/article/yes-madam",
        "notes": "",
        "screen": "NFT3",
        "time": 1732386600000,
      },
    ],
    "title": "Yes, Madam!",
    "url": "https://whatson.bfi.org.uk/Online/article/yes-madam",
  },
]
`;
